src/program.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  Keypair,
+  Connection,
+  BpfLoader,
+  BPF_LOADER_PROGRAM_ID,
+  SystemProgram,
+  Transaction,
+  sendAndConfirmTransaction,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import { ethers } from 'ethers';
+import { ContractDeployResult } from 'src';
+
+import { Contract, ContractDeployOptions } from './contract';
+import {
+  LogsParser,
+  LogCallback,
+  EventCallback,
+  parseTxError,
+  parseTxLogs,
+} from './logs';
+
+export class Program {
+  private logs: LogsParser;
+
+  /**
+   * Load a new Solang program
+   *
+   * @param connection    Solana connection
+   * @param payerAccount  Payer pubkey
+   * @param so            Solang program build bundle
+   * @returns             The program
+   */
+  static async deploy(
+    connection: Connection,
+    payerAccount: Keypair,
+    so: Buffer
+  ): Promise<Program> {
+    const programAccount = Keypair.generate();
+
+    await BpfLoader.load(
+      connection,
+      payerAccount,
+      programAccount,
+      so,
+      BPF_LOADER_PROGRAM_ID
+    );
+
+    return new Program(connection, payerAccount, programAccount);
+  }
+
+  /**
+   *
+   * Load a deployed Solang program
+   *
+   * @param connection      Solana connection
+   * @param payerAccount    Payer pubkey
+   * @param programAccount  The program's pubkey
+   * @param so              Solang program build bundle
+   * @returns               The program
+   */
+  static async get(
+    connection: Connection,
+    payerAccount: Keypair,
+    programAccount: Keypair,
+    so: Buffer
+  ): Promise<Program> {
+    await BpfLoader.load(
+      connection,
+      payerAccount,
+      programAccount,
+      so,
+      BPF_LOADER_PROGRAM_ID
+    );
+
+    return new Program(connection, payerAccount, programAccount);
+  }
+
+  /**
+   *
+   * @param connection      Solana connection
+   * @param payerAccount    Payer pubkey
+   * @param programAccount  The program's pubkey
+   */
+  constructor(
+    public connection: Connection,
+    public payerAccount: Keypair,
+    public programAccount: Keypair
+  ) {
+    this.logs = new LogsParser(this.programAccount.publicKey, this.connection);
+  }
+
+  /**
+   * Create a storage account
+   *
+   * @param space
+   * @returns
+   */
+  public async createStorageAccount(space: number): Promise<Keypair> {
+    const lamports = await this.connection.getMinimumBalanceForRentExemption(
+      space
+    );
+
+    let account = Keypair.generate();
+
+    const transaction = new Transaction().add(
+      SystemProgram.createAccount({
+        fromPubkey: this.payerAccount.publicKey,
+        newAccountPubkey: account.publicKey,
+        lamports,
+        space,
+        programId: this.programAccount.publicKey,
+      })
+    );
+
+    await sendAndConfirmTransaction(
+      this.connection,
+      transaction,
+      [this.payerAccount, account],
+      {
+        skipPreflight: false,
+        commitment: 'confirmed',
+        preflightCommitment: undefined,
+      }
```

Comment:
These should be options too. What makes skipping preflight verification desirable here?